<font size="+3"><strong>Pandas Introduction: Asignment </strong></font> 10/06/2022

#### <font color="red">Create new cell under each question; that's where to answer the respective question.

<font size="+2">Question 1</font>

Create a DataFrame called `df_mexico_2` using the `mexico-city-real-estate-2` CSV file.

In [ ]:
df_mexico_2 = 

In [8]:
import pandas as pd
df_mexico_2 = pd.read_csv("data/mexico-city-real-estate-2.csv")
df_mexico_2.head()

operation property_type                            place_with_parent_names  \
0      sell     apartment            |México|Distrito Federal|Benito Juárez|   
1      sell     apartment                  |México|Distrito Federal|Tlalpan|   
2      sell     apartment  |México|Distrito Federal|Álvaro Obregón|Tetelpan|   
3      sell     apartment            |México|Distrito Federal|Benito Juárez|   
4      sell     apartment            |México|Distrito Federal|Benito Juárez|   

                        lat-lon      price currency  \
0         19.375445,-99.1543144  4300500.0      MXN   
1  19.2742408542,-99.1496908665  2788000.0      MXN   
2  19.3427189674,-99.2225289345  3351600.0      MXN   
3        19.3596034,-99.1514055  2862800.0      MXN   
4        19.3953378,-99.1560855  3204800.0      MXN   

   price_aprox_local_currency  price_aprox_usd  surface_total_in_m2  \
0                  4261596.45        226578.22                  0.0   
1                  2762778.87        146889.91                  0.0   
2                  3321280.35        176584.01                  0.0   
3                  2836902.23        150830.86                 73.0   
4                  3175808.45        168849.64                  0.0   

   surface_covered_in_m2  price_usd_per_m2  price_per_m2  floor  rooms  \
0                   74.0               NaN  58114.864865    NaN    NaN   
1                  111.0               NaN  25117.117117    NaN    NaN   
2                   82.0               NaN  40873.170732    NaN    NaN   
3                   73.0       2066.176164  39216.438356    NaN    NaN   
4                   63.0               NaN  50869.841270    NaN    NaN   

   expenses                                      properati_url  
0       NaN  http://benito-juarez.properati.com.mx/l1yl_ven...  
1       NaN  http://tlalpan.properati.com.mx/l1ym_venta_dep...  
2       NaN  http://tetelpan.properati.com.mx/l1yn_venta_de...  
3       NaN  http://benito-juarez.properati.com.mx/l1yo_ven...  
4       NaN  http://benito-juarez.properati.com.mx/l1yp_ven...

<font size="+2">Question 2</font>

Inspect dataframe `df_mexico_2` using the Using info, shape and head methods

In [9]:
df_mexico_2.shape

(3791, 16)

In [10]:
df_mexico_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3791 entries, 0 to 3790
Data columns (total 16 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   operation                   3791 non-null   object 
 1   property_type               3791 non-null   object 
 2   place_with_parent_names     3791 non-null   object 
 3   lat-lon                     3392 non-null   object 
 4   price                       3781 non-null   float64
 5   currency                    3781 non-null   object 
 6   price_aprox_local_currency  3781 non-null   float64
 7   price_aprox_usd             3781 non-null   float64
 8   surface_total_in_m2         1207 non-null   float64
 9   surface_covered_in_m2       3470 non-null   float64
 10  price_usd_per_m2            974 non-null    float64
 11  price_per_m2                3423 non-null   float64
 12  floor                       126 non-null    float64
 13  rooms                       193 n

In [11]:
df_mexico_2.head()

operation property_type                            place_with_parent_names  \
0      sell     apartment            |México|Distrito Federal|Benito Juárez|   
1      sell     apartment                  |México|Distrito Federal|Tlalpan|   
2      sell     apartment  |México|Distrito Federal|Álvaro Obregón|Tetelpan|   
3      sell     apartment            |México|Distrito Federal|Benito Juárez|   
4      sell     apartment            |México|Distrito Federal|Benito Juárez|   

                        lat-lon      price currency  \
0         19.375445,-99.1543144  4300500.0      MXN   
1  19.2742408542,-99.1496908665  2788000.0      MXN   
2  19.3427189674,-99.2225289345  3351600.0      MXN   
3        19.3596034,-99.1514055  2862800.0      MXN   
4        19.3953378,-99.1560855  3204800.0      MXN   

   price_aprox_local_currency  price_aprox_usd  surface_total_in_m2  \
0                  4261596.45        226578.22                  0.0   
1                  2762778.87        146889.91                  0.0   
2                  3321280.35        176584.01                  0.0   
3                  2836902.23        150830.86                 73.0   
4                  3175808.45        168849.64                  0.0   

   surface_covered_in_m2  price_usd_per_m2  price_per_m2  floor  rooms  \
0                   74.0               NaN  58114.864865    NaN    NaN   
1                  111.0               NaN  25117.117117    NaN    NaN   
2                   82.0               NaN  40873.170732    NaN    NaN   
3                   73.0       2066.176164  39216.438356    NaN    NaN   
4                   63.0               NaN  50869.841270    NaN    NaN   

   expenses                                      properati_url  
0       NaN  http://benito-juarez.properati.com.mx/l1yl_ven...  
1       NaN  http://tlalpan.properati.com.mx/l1ym_venta_dep...  
2       NaN  http://tetelpan.properati.com.mx/l1yn_venta_de...  
3       NaN  http://benito-juarez.properati.com.mx/l1yo_ven...  
4       NaN  http://benito-juarez.properati.com.mx/l1yp_ven...

<font size="+2">Question 3</font>

What can you deduce from about dataset from `df_mexico_2` inspection?

The shape output shows that mexico_2 Dataframe -- mexico_2 has 3791 rows and 6 columns
The data in mexico_2 is both float and object types

<font size="+2">Question 4</font>

Create a DataFrame called `df_mexico_3` using the `mexico-city-real-estate-3` CSV file.

In [12]:
import pandas as pd
df_mexico_3 = pd.read_csv("data/mexico-city-real-estate-3.csv")
df_mexico_3.head()

operation property_type                            place_with_parent_names  \
0      sell     apartment  |México|Distrito Federal|Tlalpan|Fuentes del P...   
1      sell     apartment      |México|Distrito Federal|Venustiano Carranza|   
2      sell         house                |México|Distrito Federal|Iztacalco|   
3      sell     apartment      |México|Distrito Federal|Venustiano Carranza|   
4      sell     apartment            |México|Distrito Federal|Benito Juárez|   

                lat-lon      price currency  price_aprox_local_currency  \
0  19.302422,-99.230829  1962000.0      MXN                  1944251.19   
1  19.428152,-99.106194  1275000.0      MXN                  1263465.87   
2  19.404611,-99.069557  1866667.0      MXN                  1849780.61   
3  19.428152,-99.106194  1275000.0      MXN                  1263465.87   
4    19.3653,-99.148217  1210000.0      MXN                  1199053.91   

   price_aprox_usd  surface_total_in_m2  surface_covered_in_m2  \
0        103370.88                  NaN                  100.0   
1         67175.26                  NaN                   68.0   
2         98348.12                  NaN                   95.0   
3         67175.26                  NaN                   68.0   
4         63750.64                  NaN                   67.0   

   price_usd_per_m2  price_per_m2  floor  rooms  expenses  \
0               NaN  19620.000000    NaN    NaN       NaN   
1               NaN  18750.000000    NaN    NaN       NaN   
2               NaN  19649.126316    NaN    NaN       NaN   
3               NaN  18750.000000    NaN    NaN       NaN   
4               NaN  18059.701493    NaN    NaN       NaN   

                                       properati_url  
0  http://fuentes-del-pedregal.properati.com.mx/o...  
1  http://venustiano-carranza.properati.com.mx/o4...  
2  http://iztacalco.properati.com.mx/o460_venta_c...  
3  http://venustiano-carranza.properati.com.mx/o4...  
4  http://benito-juarez.properati.com.mx/o463_ven...

<font size="+2">Question 5</font>

Merge/Concatenate `df_mexico_2` and `df_mexico_3` and save it to a new dataframe `df_new`

In [4]:
import pandas as pd

In [5]:
df_mexico_2 = pd.read_csv("data/mexico-city-real-estate-2.csv")
df_mexico_3 = pd.read_csv("data/mexico-city-real-estate-3.csv")
print("df_mexico_2 shape:", df_mexico_2.shape)
print("df_mexico_3 shape:", df_mexico_3.shape)

df_mexico_2 shape: (3791, 16)
df_mexico_3 shape: (3791, 16)


In [6]:
df_new = pd.concat([df_mexico_2, df_mexico_3])
print("df_new shape:", df_new.shape)
df_new.head()

df_new shape: (7582, 16)


operation property_type                            place_with_parent_names  \
0      sell     apartment            |México|Distrito Federal|Benito Juárez|   
1      sell     apartment                  |México|Distrito Federal|Tlalpan|   
2      sell     apartment  |México|Distrito Federal|Álvaro Obregón|Tetelpan|   
3      sell     apartment            |México|Distrito Federal|Benito Juárez|   
4      sell     apartment            |México|Distrito Federal|Benito Juárez|   

                        lat-lon      price currency  \
0         19.375445,-99.1543144  4300500.0      MXN   
1  19.2742408542,-99.1496908665  2788000.0      MXN   
2  19.3427189674,-99.2225289345  3351600.0      MXN   
3        19.3596034,-99.1514055  2862800.0      MXN   
4        19.3953378,-99.1560855  3204800.0      MXN   

   price_aprox_local_currency  price_aprox_usd  surface_total_in_m2  \
0                  4261596.45        226578.22                  0.0   
1                  2762778.87        146889.91                  0.0   
2                  3321280.35        176584.01                  0.0   
3                  2836902.23        150830.86                 73.0   
4                  3175808.45        168849.64                  0.0   

   surface_covered_in_m2  price_usd_per_m2  price_per_m2  floor  rooms  \
0                   74.0               NaN  58114.864865    NaN    NaN   
1                  111.0               NaN  25117.117117    NaN    NaN   
2                   82.0               NaN  40873.170732    NaN    NaN   
3                   73.0       2066.176164  39216.438356    NaN    NaN   
4                   63.0               NaN  50869.841270    NaN    NaN   

   expenses                                      properati_url  
0       NaN  http://benito-juarez.properati.com.mx/l1yl_ven...  
1       NaN  http://tlalpan.properati.com.mx/l1ym_venta_dep...  
2       NaN  http://tetelpan.properati.com.mx/l1yn_venta_de...  
3       NaN  http://benito-juarez.properati.com.mx/l1yo_ven...  
4       NaN  http://benito-juarez.properati.com.mx/l1yp_ven...

<font size="+2">Question 6</font>

Delete `rooms` and `floor` columns in `df_new`.

In [ ]:
df_new

In [13]:
df_new = df_new.drop(["rooms", "floor"], axis=1)
df_new.head()

operation property_type                            place_with_parent_names  \
0      sell     apartment            |México|Distrito Federal|Benito Juárez|   
1      sell     apartment                  |México|Distrito Federal|Tlalpan|   
2      sell     apartment  |México|Distrito Federal|Álvaro Obregón|Tetelpan|   
3      sell     apartment            |México|Distrito Federal|Benito Juárez|   
4      sell     apartment            |México|Distrito Federal|Benito Juárez|   

                        lat-lon      price currency  \
0         19.375445,-99.1543144  4300500.0      MXN   
1  19.2742408542,-99.1496908665  2788000.0      MXN   
2  19.3427189674,-99.2225289345  3351600.0      MXN   
3        19.3596034,-99.1514055  2862800.0      MXN   
4        19.3953378,-99.1560855  3204800.0      MXN   

   price_aprox_local_currency  price_aprox_usd  surface_total_in_m2  \
0                  4261596.45        226578.22                  0.0   
1                  2762778.87        146889.91                  0.0   
2                  3321280.35        176584.01                  0.0   
3                  2836902.23        150830.86                 73.0   
4                  3175808.45        168849.64                  0.0   

   surface_covered_in_m2  price_usd_per_m2  price_per_m2  expenses  \
0                   74.0               NaN  58114.864865       NaN   
1                  111.0               NaN  25117.117117       NaN   
2                   82.0               NaN  40873.170732       NaN   
3                   73.0       2066.176164  39216.438356       NaN   
4                   63.0               NaN  50869.841270       NaN   

                                       properati_url  
0  http://benito-juarez.properati.com.mx/l1yl_ven...  
1  http://tlalpan.properati.com.mx/l1ym_venta_dep...  
2  http://tetelpan.properati.com.mx/l1yn_venta_de...  
3  http://benito-juarez.properati.com.mx/l1yo_ven...  
4  http://benito-juarez.properati.com.mx/l1yp_ven...

<font size="+2">Question 7</font>

Create a column `final_price` in `df_new`. The value of `final_price` is `price` multiplied by `surface_covered_in_m2` from `df_new`.

In [ ]:
df_new

In [15]:
df_new["final_price"] = df_new["price"] * df_new["surface_covered_in_m2"]
df_new.head()

operation property_type                            place_with_parent_names  \
0      sell     apartment            |México|Distrito Federal|Benito Juárez|   
1      sell     apartment                  |México|Distrito Federal|Tlalpan|   
2      sell     apartment  |México|Distrito Federal|Álvaro Obregón|Tetelpan|   
3      sell     apartment            |México|Distrito Federal|Benito Juárez|   
4      sell     apartment            |México|Distrito Federal|Benito Juárez|   

                        lat-lon      price currency  \
0         19.375445,-99.1543144  4300500.0      MXN   
1  19.2742408542,-99.1496908665  2788000.0      MXN   
2  19.3427189674,-99.2225289345  3351600.0      MXN   
3        19.3596034,-99.1514055  2862800.0      MXN   
4        19.3953378,-99.1560855  3204800.0      MXN   

   price_aprox_local_currency  price_aprox_usd  surface_total_in_m2  \
0                  4261596.45        226578.22                  0.0   
1                  2762778.87        146889.91                  0.0   
2                  3321280.35        176584.01                  0.0   
3                  2836902.23        150830.86                 73.0   
4                  3175808.45        168849.64                  0.0   

   surface_covered_in_m2  price_usd_per_m2  price_per_m2  expenses  \
0                   74.0               NaN  58114.864865       NaN   
1                  111.0               NaN  25117.117117       NaN   
2                   82.0               NaN  40873.170732       NaN   
3                   73.0       2066.176164  39216.438356       NaN   
4                   63.0               NaN  50869.841270       NaN   

                                       properati_url  final_price  
0  http://benito-juarez.properati.com.mx/l1yl_ven...  318237000.0  
1  http://tlalpan.properati.com.mx/l1ym_venta_dep...  309468000.0  
2  http://tetelpan.properati.com.mx/l1yn_venta_de...  274831200.0  
3  http://benito-juarez.properati.com.mx/l1yo_ven...  208984400.0  
4  http://benito-juarez.properati.com.mx/l1yp_ven...  201902400.0

<font size="+2">Question 8</font>

Save `df_new` as a new csv file `'mexico23.csv'`

In [17]:
df_new = df_new.head()
df_new.to_csv("data/mexico23.csv", index=False)

In [18]:
df_new.to_excel("mexico23.xlsx")
print("file saved sucessfully")

file saved sucessfully


<font size="+2">Note:</font>

Create new cells where necessary and type all answers in this notebook.

You are to submit a copy of this workbook and `mexico23.csv`

Submission Date: 12/06/2022

Submission email: dsclass@dreamboxtech.com